### PR TITLE
readbytes and readall, replaced by read and readstring per #14660

### PR DESCRIPTION
**Pull request by samoconnor**
Hi, @kmsquire, @nalimilan, re https://github.com/JuliaLang/GZip.jl/pull/45#issuecomment-173524469: I've made an attempt. I haven't looked at Compat.jl before, so I'm not sure if I'm doing it right.

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -717,4 +717,27 @@ if VERSION < v"0.5.0-dev+1946"
     export supertype
 end
 
+
+if VERSION < v"0.5.0-dev+2228"
+    const readall = readstring
+    const readbytes = read
+    export readall
+    export readbytes
```
Comment (yuyichao) [inline src/Compat.jl]: Looks like it's at least reversed. Also I believe only `readstring` is a new name. Other ones shouldn't be overwirtten.

**Comment by yuyichao** [comment]
And please use a more descriptive commit and PR title.

File: src/Compat.jl (hunk 1 above)
Comment (samoconnor) [inline src/Compat.jl]: thanks for the tips @yuyichao!

**Comment by jrevels** [comment]
@samoconnor Could you rebase against the current master and add some tests? Otherwise, LGTM. This will fix a lot of annoying deprecation warnings for web-related stuff.

**Comment by samoconnor** [comment]
@jrevels, done

**Comment by tkelman** [comment]
`ERROR: `mktempdir` has no method matching mktempdir(::Function)` on 0.3 (that could be added as a new feature here in a separate PR though), and `ERROR: LoadError: MethodError: Cannot `convert` an object of type Array{Any,1} to an object of type UTF8String` on nightly.

**Comment by samoconnor** [comment]
Hi @tkelman,
I don't have a nightly build to test this against, but it works with the 0.5 build I do have (Version 0.5.0-dev+2236 (2016-01-21 23:04 UTC)).
Is this possibly a type-inference regression in nightly?

This is the code that is failing:
```julia
        for text in [
            old_text,
            UTF8String(['A' + i % 52 for i in 1:(div(Base.SZ_UNBUFFERED_IO,2))]),
            UTF8String(['A' + i % 52 for i in 1:(    Base.SZ_UNBUFFERED_IO -1)]),
            UTF8String(['A' + i % 52 for i in 1:(    Base.SZ_UNBUFFERED_IO   )]),
            UTF8String(['A' + i % 52 for i in 1:(    Base.SZ_UNBUFFERED_IO +1)])
        ]
```

It looks to me that the compiler ought to be able to tell that those are `Char` arrays:
`Char` + `Int` % `Int` = `Char`

**Comment by tkelman** [comment]
A type inference issue seems likely. Adding the `Char[...]` element type might work around the issue for now, but this could be a bug worth opening on base as well.

**Comment by samoconnor** [comment]
See https://github.com/JuliaLang/julia/commit/639621859863609c5f3abbc2ed75c675695b3693#diff-435745fdfa02a0dc4f3b747ba977af11L180

File: test/runtests.jl [hunk 2]
```diff
@@ -560,3 +560,199 @@ Base.remote_do(() -> true, 1) # Doesn't return anything so cannot be `@test`ed b
 @test "1234" == @compat withenv("_TESTVAR" => 1234) do
     return ENV["_TESTVAR"]
 end
+
+# https://github.com/JuliaLang/julia/pull/14660
+mktempdir() do dir
```
Comment (stevengj) [inline test/runtests.jl]: This relies on JuliaLang/julia#9017, which was introduced in Julia 0.4.

Probably we should have a Compat shim for that `mktempdir` feature on 0.3 as well.

Comment (stevengj) [inline test/runtests.jl]: Oh, I see @tkelman commented on this above.

**Comment by malmaud** [comment]
How's this PR doing? Can I help? Would love to see Requests.jl green again.

**Comment by tkelman** [comment]
A separate addition of `mktempdir` should probably be the only thing this needs. We could do that in a separate PR, once that's good to merge we can just restart the CI builds here and they should pass.

**Comment by malmaud** [comment]
OK, I'll do that today.

**Comment by samoconnor** [comment]
Hi @malmaud, that would me much appreciated.
I'm afraid I'm not at all familiar with `Compat.jl`, so this PR is just my best attempt.
I do all my work against v0.4.
I occasionally build a v0.5 environment if I'm trying to submit a bug fix, but I don't have any setup for testing against v0.3.
If you find anything else is broken in this PR please feel free to hijack my branch into your own fork and fix it up before merging with master.

**Comment by tkelman** [comment]
You can download and extract a tarball binary to test against 0.3. It couldn't be any easier.

**Comment by malmaud** [comment]
@samoconnor OK, I'll take you up on that

**Comment by tkelman** [comment]
thank @malmaud, do open a new pr so appveyor runs

**Pull request closed without merging**